src/System/Router/renderServerApp.tsx

```diff
@@ -80,7 +84,51 @@ export const renderServerApp = ({
     sd: sharify.data,
   }
 
+  // Preconnect or preload associated links
+  setEarlyHintsResponseHeaders({
+    req,
+    res,
+    imageLinkHeaders,
+    scriptLinkHeaders,
+  })
+
   const statusCode = getENV("statusCode") ?? code
 
   res.status(statusCode).render(`${PUBLIC_DIR}/html.ejs`, options)
 }
+
+/**
+ * Link headers allow 103: Early Hints to be sent to the client (by Cloudflare).
+ * These should correspond to preconnect and preload items from html.ejs.
+ * See: https://developers.cloudflare.com/cache/advanced-configuration/early-hints/
+ */
+interface SetEarlyHintsResponseHeadersProps {
+  imageLinkHeaders: string[]
+  scriptLinkHeaders: string[]
+  req: ArtsyRequest
+  res: ArtsyResponse
+}
+
+const setEarlyHintsResponseHeaders = ({
+  imageLinkHeaders,
+  scriptLinkHeaders,
+  res,
+}: SetEarlyHintsResponseHeadersProps) => {
+  try {
+    if (!res.headersSent) {
+      res.header("Link", [
+        `<${CDN_URL}>; rel=preconnect; crossorigin`,
+        `<${GEMINI_CLOUDFRONT_URL}>; rel=preconnect; crossorigin`,
+        `<${WEBFONT_URL}>; rel=preconnect; crossorigin`,
```

Comment:
Thought: Now that we're preloading js assets and fonts and a main image, I wonder if we even need these preconnects (that likely refer to the same domains). We can try removing them later.